stream-chat-android-client/src/main/java/io/getstream/chat/android/client/socket/ChatSocket.kt

```diff
@@ -297,9 +302,13 @@ internal open class ChatSocket constructor(
 
     private fun callListeners(call: (SocketListener) -> Unit) {
         synchronized(listeners) {
-            listeners.forEach { listener ->
-                eventUiHandler.post { call(listener) }
-            }
+            listeners.forEach(call)
+        }
+    }
+
+    private fun callListenersOnUiThread(call: (SocketListener) -> Unit) {
+        eventUiHandler.post {
+            callListeners(call)
```

Comment:
Why some of the events are notified on the UI thread and other no?

Reply:
It is not necessary indeed.

Reply:
Sorry, it is actually necessary. Using `eventUiHandler.post {` forces the listeners to be async. This can't happen in the disconnection, because that would cause that we return `Result.success` before the socket is completed disconnected... the listeners would run after the disconnection is complete, which would allow users to try to connect before they are allowe to.

Reply:
Instead of using `eventUiHandler.post {}`, you can use `coroutineScope.launch(DispatchersProvider.Main) { }` and, because we are cancelling all coroutineScopes of our SDK, it will be cancelled when `ChatClient.disconnect()` is called.

Reply:
Indeed, this works. I tested in several situations and the disconnection always happens after this events are completed.